api/src/controllers/assets.ts

```diff
@@ -140,11 +140,11 @@ router.get(
 
 		if (req.headers.range) {
 			// substring 6 = "bytes="
-			const rangeParts = req.headers.range.substring(6).split('-');
+			const rangeParts = /bytes=(.*)-(.*)/.exec(req.headers.range);
```

Comment:
```suggestion
			const rangeParts = /bytes=([0-9]*)-([0-9]*)/.exec(req.headers.range);
```

I prefer a more strict regex

Reply:
Yes, I also I thought about this one, but I think we should catch what's inside the header then in the following lines we alert about misconfigured `Range` header:
https://github.com/directus/directus/blob/c19d163759f8e5e4c3490560802a42f76361554f/api/src/controllers/assets.ts#L145-L152

Reply:
More worried about the wildcard. Suggested regex eliminates the need for a forward check on `-` character.

Reply:
I believe this no longer is true since I made that regex lazy. But can you confirm?

Reply:
`/bytes=(.*)-(.*)/` matches things like

```javascript
'bytes=insert_malformed_data_here-insert_more_malformed_data_here'
```

`/bytes=([0-9]*)-([0-9])*)/` only accepts digits

Reply:
I get your point, but we need to tell the user about the requested range here, in case of failure:
https://github.com/directus/directus/blob/55622a6613ea36048cf552376237fc81a2521b07/api/src/exceptions/range-not-satisfiable.ts#L7

That's why we catch everything, so user/developer can see what's going on (like we did).

Reply:
@wrynegade FWIW, `/bytes=([0-9]*)-([0-9])*)/` has a typo, believe it should be `/bytes=([0-9]*)-([0-9]*)/`

Reply:
We should update this too. Especially since `range.start` and `range.end` are Directus generated values, there is a chance that `${range.start}-${range.end}` is not the same as the original value. Perhaps we need to include the original header value in the `Range` type?

Reply:
> @wrynegade FWIW, `/bytes=([0-9]*)-([0-9])*)/` has a typo, believe it should be `/bytes=([0-9]*)-([0-9]*)/`

haha thanks; updated the comment

Reply:
Regex and exception message updated. 👍